File: Day1/day1.py
import heapq

def getElves(input: str) -> list[int]:
    return[
        sum(map(int, lines.split()))
        for lines in input.split("\n\n")
    ]

def part1(elves: list[int]) -> int:
    return max(elves)

def part2(elves: list[int]) -> int:
    return sum(heapq.nlargest(3, elves))

if __name__ == "__main__":
    with open("/Users/alex.waring/code/AoC2022/Day1/input.txt") as file:
        print(part1(getElves(file.read())))
    with open("/Users/alex.waring/code/AoC2022/Day1/input.txt") as file:
        print(part2(getElves(file.read())))

File: Day1/day1.go
package main
import (
	"fmt"
	"os"
	"strings"
	"strconv"
)

func check(e error) {
	if e != nil {
		panic(e)
	}
}

func sum(array []string) int {  
	result := 0  
	for _, v := range array {  
		value, err := strconv.Atoi(v)
		check(err)
		result += value
	}  
	return result  
}

func findmax(array []int) int {
	max := 0
	for _, v := range array {
		if v > max {
			max = v
		}
	}
	return max
}

func findmax3sum(array []int) int {
	max := [3]int{0, 0, 0}
	for _, v := range array {
		if v > max[0] {
			max[0] = v
		} else if v > max[1] {
			max[1] = v
		} else if v > max[2] {
			max[2] = v
		}
	}
	result := 0
	for _, v := range max {
		result += v
	}
	return result
}

func getElves(input string) []int {
	elves := strings.Split(input, "\n\n")
	elvessum := []int{}

	for _, value := range elves {
		total := sum(strings.Split(value, "\n"))
		elvessum = append(elvessum, total)
	}

	return elvessum
}

func main() {
	input, err := os.ReadFile("input.txt")
	check(err)
	output := getElves(string(input))
	check(err)
	fmt.Println(findmax(output))
	fmt.Println(findmax3sum(output))
}